### ISSUE TITLE
Profile switch validation

### ISSUE THREAD

**Issue by yalis** [mikepenz/MaterialDrawer#1206]
Is it possible to execute some validations actions on the profile before effectively switching to the selected profile?

Our actual use case requires the user input a valid pin number before being able to switch profile.

Thanks

**Comment by mikepenz** [comment]
@yalis as of now there is no `pre` click listener which is called before changing.

**Comment by mikepenz** [comment]
@yalis you can prevent the selection within the list if you set a `preClickListener` to the `Adapter`. Make sure you return false for normal items, and true for `ProfileDrawerItems` 

```java
result.getAdapter().withOnPreClickListener(new FastAdapter.OnClickListener<IDrawerItem>() {
            @Override
            public boolean onClick(View v, IAdapter<IDrawerItem> adapter, IDrawerItem item, int position) {
                return true;
            }
        });
```

For the selection via the small icons in the header there is no possibility at the moment. I will look into it now.

**Comment by yalis** [comment]
Thanks @mikepenz for the response.

I tested your solution and it work good.

Before reading the answer I added a Pre Profile Swtich Listener 

https://github.com/yalis/MaterialDrawer/commit/1fd6c065d4cd7fb26f8bda1abb4cc9dacf5d4ad5

From the library user point of view it looks easier to use this functionality.

`
.withOnAccountHeaderPreProfileSwitchListener(new AccountHeader.OnAccountHeaderPreProfileChangeListener() {
                    @Override
                    public boolean onPreProfileChange(View view, IProfile profile, boolean current) {
                              return true:
                    }
`
